test/relay.test.js

```diff
@@ -0,0 +1,350 @@
+'use strict';
+
+var chai = require('chai')
+  , expect = chai.expect
+  , resolver = require('../src/resolver')
+  , helper = require('./helper')
+  , sequelize = helper.sequelize
+  , Sequelize = require('sequelize')
+  , Promise = helper.Promise
+  , attributeFields = require('../src/attributeFields');
+
+import {
+  GraphQLString,
+  GraphQLInt,
+  GraphQLFloat,
+  GraphQLNonNull,
+  GraphQLBoolean,
+  GraphQLEnumType,
+  GraphQLList,
+  GraphQLObjectType,
+  GraphQLSchema,
+  graphql
+} from 'graphql';
+
+import {
+  sequelizeNodeInterface
+} from '../src/relay';
+
+import {
+  globalIdField,
+  toGlobalId,
+  connectionDefinitions,
+  connectionArgs,
+  connectionFromArray
+} from 'graphql-relay';
+
+function generateTask(id) {
+  return {
+    id: id,
+    name: Math.random().toString()
+  }
+}
+
+describe('relay', function () {
+  var User
+    , Task
+    , userType
+    , taskType
+    , taskConnection
+    , nodeInterface
+    , Project
+    , projectType
+    , userConnection
+    , nodeField
+    , schema;
+
+  before(function () {
+    sequelize.modelManager.models = [];
+    sequelize.models = {};
+    User = sequelize.define('user', {
+      name: {
+        type: Sequelize.STRING
+      }
+    }, {
+      timestamps: false
+    });
+
+    Task = sequelize.define('task', {
+      name: {
+        type: Sequelize.STRING
+      }
+    }, {
+      timestamps: false
+    });
+
+    Project = sequelize.define('project', {
+      name: {
+        type: Sequelize.STRING
+      }
+    }, {
+      timestamps: false
+    });
+
+    User.Tasks = User.hasMany(Task, {as: 'tasks'});
+    Project.Users = Project.hasMany(User, {as: 'users'});
+
+    var node = sequelizeNodeInterface(sequelize, (obj) => {
+      switch(obj.Model.options.name.singular) {
+        case 'user':
+          return userType;
+          break;
+        case 'project':
+          return projectType;
+          break;
+        case 'task':
+          return taskType;
+          break;
+        default:
+          return null;
+      }
+    });
```

Comment:
`sequelizeNodeInterface` could also return a `typeMap` property which the user needs to fill out with `modelName => type` mappings.

```js
let {
  nodeInterface,
  nodeField,
  nodeTyperMapper
} = sequelizeNodeInterface(sequelize);

nodeTypeMapper[User.name] = userType;
```

With the user providing a resolve function themselves they could just as well do:

```
nodeDefinitions(idFetcher, (obj) => { ... });
```

Providing a helper in that case seems a little pointless.